compiler/lib/flow.ml

```diff
@@ -237,6 +238,7 @@ let program_escape defs known_origins { blocks; _ } =
       List.iter block.body ~f:(fun i ->
           match i with
           | Let (x, e) -> expr_escape st x e
+          | Assign (x, _) -> possibly_mutable.(Var.idx x) <- true
```

Comment:
We want `maybe_unknown` to be true (that is, we do not know all possible values of `x`), not `possibly_mutable` to be true (the fields of the values of `x` may be modified).

Reply:
Or if we have `Assign (x, y)`, `y` should be a known origin of `x`